src/Sarif/Core/Result.cs

```diff
@@ -73,18 +73,21 @@ public ReportingDescriptor GetRule(Run run = null)
                     return GetRuleByIndex(rules, this.Rule.Index);
                 }
 
-                // Look up by this.RuleDescriptor, Guid, if present
-                if (!string.IsNullOrEmpty(this.Rule?.Guid))
+                // Look up by this.RuleDescriptor.Guid, if present
+                if (!string.IsNullOrEmpty(this.Rule?.Guid) && rules != null)
                 {
-                    if (rules != null)
-                    {
-                        foreach (ReportingDescriptor rule in rules)
-                        {
-                            if (rule.Guid == this.Rule.Guid) { return rule; }
-                        }
-
-                        throw new ArgumentException($"ReportingDescriptorReference referred to Guid {this.Rule.Guid}, which was not found in toolComponent.Rules.");
-                    }
+                    ReportingDescriptor rule = component.GetRuleByGuid(this.Rule.Guid);
+                    if (rule != null) { return rule; }
+                    throw new ArgumentException($"ReportingDescriptorReference referred to Guid {this.Rule.Guid}, which was not found in toolComponent.Rules.");
```

Comment:
>throw new ArgumentException($"ReportingDescriptorReference referred to Guid {this.Rule.Guid}, which was not found in toolComponent.Rules.") [](start = 20, length = 139)

I don't think we should throw because the SARIF is invalid. Just don't return a `ReportingDescriptor` if you can't find one (or, in this case, fall through to the next block). Comment applies to the ruleId lookup as well.

Reply:
Fixed